detectors/LambdaDetector.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Detectors\Aws;
+
+use OpenTelemetry\Sdk\Resource\ResourceConstants;
+use OpenTelemetry\Sdk\Resource\ResourceInfo;
+use OpenTelemetry\Sdk\Trace\Attributes;
+
+/**
+ * The AwsLambdaDetector can be used to detect if a process is running in AWS Lambda
+ * and return a {@link Resource} populated with data about the environment.
+ * Returns an empty Resource if detection fails.
+ */
+class LambdaDetector
+{
+    private const LAMBDA_NAME_ENV = 'AWS_LAMBDA_FUNCTION_NAME';
+    private const LAMBDA_VERSION_ENV = 'AWS_LAMBDA_FUNCTION_VERSION';
+    private const AWS_REGION_ENV = 'AWS_REGION';
+    private const CLOUD_PROVIDER = 'aws';
+
+    public function detect(): ResourceInfo
+    {
+        $lambdaName = getenv(self::LAMBDA_NAME_ENV);
+        $functionVersion = getenv(self::LAMBDA_VERSION_ENV) ;
+        $awsRegion = getenv(self::AWS_REGION_ENV);
+        
+        // The following bitwise functions are created because
```

Comment:
I don't see any bitwise functions here, only a pile of ternary statements.

Reply:
Ope, I confused the names! will fix that comment now